charts/opentelemetry-operator/templates/_helpers.tpl

```diff
@@ -23,6 +23,17 @@ If release name contains chart name it will be used as a full name.
 {{- end }}
 {{- end }}
 
+{{/*
+Allow the release namespace to be overridden
+*/}}
+{{- define "opentelemetry-operator.namespace" -}}
+  {{- if .Values.namespaceOverride -}}
+    {{- .Values.namespaceOverride -}}
+  {{- else -}}
+    {{- .Release.Namespace -}}
+  {{- end -}}
+{{- end -}}
```

Comment:
Namespace names have a [char size limit of 63](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names) I've seen users hit before, I'd suggest truncating the value based on the limit with something like this.
`{{- default .Release.Namespace .Values.namespaceOverride | trunc 63 | trimSuffix "-" -}}
`